### DISCUSSION TITLE
Vatlayer taxes overridden by order price recalculation in 3.6

### DISCUSSION THREAD

**Discussion by carlosa54** [saleor/saleor#10431] [Q&A] [answered]
Hi 👋 

I noticed that the Vatlayer taxes for a checkout/order in status `draft` or `unconfirmed` will get overridden by the new logic of recalculating the checkout/order prices.

Once the order is created the taxes are applied correctly but if someone fetches that order the following will get called

https://github.com/saleor/saleor/blob/2b8f5b28163e0618576fa45a96d195411ade3c9e/saleor/order/calculations.py#L119-L132
This will call the `_recalculate_order_prices`
https://github.com/saleor/saleor/blob/2b8f5b28163e0618576fa45a96d195411ade3c9e/saleor/order/calculations.py#L146 and since the Vatlayer plugin doesn't implement the plugin functions that runs inside `_recalculate_order_prices` (`calculate_order_line_unit` and `calculate_order_line_total`) this will fallback to use the default value without taxes overriding the previously Vatlayer calculated totals.

This will probably be fixed in 3.7 when https://github.com/saleor/saleor/pull/9784 gets merged but I wanted to let you know that 3.6  has that bug when using Vatlayer

@fowczarek @korycins

**Comment by korycins** [comment, marked as answer]
Hey @carlosa54 ,
Thanks a lot for raising this and giving so much context ❤️ ! As you said there is a missing implementation of these methods for Vatlayer plugin. 

BTW. I just saw that you have your own tax plugin. Nice work! I would propose you to getting familiar with our new sync webhook tax flow: https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/tax-webhooks 
It should be easier for you to maintain your code as you don't need to worry about any changes made inside a Saleor's core code.
You can check Saleor-Taxjar app as an example: https://github.com/saleor/saleor-taxjar (It is still missing some logic part but I think it is worth checking)

**Reply by carlosa54** [reply]
Nice 👍  thanks for the references. I think I might update my plugin and implement the new `get_taxes_for_order` and `get_taxes_for_checkout` methods in the meantime

**Comment by carlosa54** [comment]
@korycins I just updated the [plugin](https://github.com/carlosa54/saleor-flat-tax-plugin/releases/tag/v0.5.1) implementing the `get_taxes_for_order` method to fix the regression when recalculating orders. I left the other plugin methods as well because for my use case I also want the plugin to take care of taxes in product prices without the need to update my frontend. (Not only in the checkout and order process)

Thanks again for your quick response
